exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -149,6 +174,29 @@ func (r *Reconciler) reconcileDelete(_ context.Context, extensionConfig *runtime
 	return ctrl.Result{}, nil
 }
 
+// secretToExtensionConfig maps a secret to ExtensionConfigs to reconcile them on updates of the secrets.
+func (r *Reconciler) secretToExtensionConfig(secret client.Object) []reconcile.Request {
```

Comment:
I would suggest to implement the index feature builtin into controller-runtime instead of implementing our own.

I didn't suggest this before because I thought we don't have a performance issue here.
Problem of using the index is that we have to write the test with envtest as fake client doesn't support it.

Reply:
I could do that tomorrow in this or in a follow-up PR 😀 

I think migrating to envtest would also be ok and not that hard. The other tests in that package also use envtest.

Reply:
kk, let's add this in the tracking issue, then we can decide when to do this

Reply:
Done